src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -654,12 +668,26 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
     }
 }
 
+pub(super) struct IncludingDowncast(bool);
```

Comment:
It's usually a bit more idiomatic to use an enum with the possible states, but this is still an improvement in readibility over a naked Boolean.

Reply:
Yeah, I think I prefer an enum, was just following @nikomatsakis suggestion. But I'm fine with either.